vyos/vyos-1x
# validators: T4472: Add validator ipv4-range-go in golang

## Change Summary

Go validators decrease the validation time to compares with
bash + ipaddrcheck --something or python3
As PoC we decrease the validation time for `ipv4-range` (10K checks)
from 48 seconds to 27 seconds


## Types of changes


## Related Task(s)

* https://phabricator.vyos.net/T4472

## Component(s) name


## Proposed changes


## How to test
Check validations:
Standard `ipv4-range`
```
vyos@r14# time sudo sh -c ' for ((n=0;n<10000;n++)); do  /usr/libexec/vyos/validate-value  --exec /usr/libexec/vyos/validators/ipv4-range --value "192.0.2.1-192.0.2.100"; done'

real	0m48.825s
user	0m26.534s
sys	0m23.904s
[edit]
vyos@r14#
```
The same validator in `go` 
```
vyos@r14# time sudo sh -c ' for ((n=0;n<10000;n++)); do  /usr/libexec/vyos/validate-value  --exec /usr/libexec/vyos/validators/ipv4-range-go --value "192.0.2.1-192.0.2.100"; done'

real	0m27.069s
user	0m12.814s
sys	0m14.784s
[edit]
vyos@r14#
```

## Checklist:
